api/src/db/legacy_mixins/forecast_mixin.py

```diff
@@ -0,0 +1,158 @@
+#
```

Comment:
Nit: most of the code uses singular, should we change `/legacy_mixins/` → `/legacy_mixin/`